### PR TITLE
Port test suite from nose to pytest

### PR DESCRIPTION
**Pull request by mcepl**
This is just minimal set of patches to make nose go away. Clean up, and reinserting tests to your CI is left as an exercise for the reader.

Edit by @bblommers : PR is now ready.
Biggest changes:
 - Use `pytest.raises` (instead of `assert_raises`)
 - When asserting these errors, use `ex.value.response` (instead of ex.exception.response`)
 - Use the  build-in `pytest.mark.parametrize` (instead of the external `parameterized` dependency)

### CODE REVIEW

**Comment by mcepl** [comment]
Probably replaces #3354.

**Comment by bblommers** [comment]
@spulec Is this a route we want to take?
Nose is no longer maintained, yes, but I feel it works fine as is - there aren't any bugs/features in nose that we need fixed.

**Comment by mcepl** [comment]
> @spulec Is this a route we want to take?
> Nose is no longer maintained, yes, but I feel it works fine as is - there aren't any bugs/features in nose that we need fixed.

Except it will stop working with Python 3.9 (or 3.10, not sure which one).

**Comment by spulec** [comment]
Yeah, it does seem like we need to eventually move off Nose.

I'm good to merge if we can get the conflicts fixed. Apologies for the slow response leading to more branch conflicts.

**Comment by bblommers** [comment]
> Except it will stop working with Python 3.9 https://tirkarthi.github.io/programming/2020/01/27/python-39-changes.html.

I wasn't aware of this! I see you're already merging in master @mcepl - I'll hold off merging anything else until this is done.
Let me know if I can help in any way.

**Comment by mcepl** [comment]
> I see you're already merging in master @mcepl - I'll hold off merging anything else until this is done.

??? That was at me? I certainly cannot merge here, I am just humbly submitting this PR for your gracious consideration.

**Comment by bblommers** [comment]
I meant merging master into  your own branch:
`mcepl force-pushed the mcepl:denose branch from 9a3ac6e to 5f31255 5 hours ago`

**Comment by mcepl** [comment]
> I meant merging master into your own branch:
> `mcepl force-pushed the mcepl:denose branch from 9a3ac6e to 5f31255 5 hours ago`

No, I did `git rebase master` on my local branch `denose` and then I had to force push it, because all commits were changed.

**Comment by bblommers** [comment]
Looks like Travis is not quite happy with the changes:
https://travis-ci.org/github/spulec/moto/builds/742642006

You might need to run `black` as well, to keep the linter happy.

**Comment by bblommers** [comment]
@mcepl Hope you don't mind - I took the liberty to make a few changes as well

**Comment by mcepl** [comment]
I think this is the moment when I spent too much time on this and when I did all I could do. Finishing this pull request is left as the exercise for other contributors.

**Comment by bblommers** [comment]
Sounds good @mcepl. I'll pick it up from here on.

Thanks for all your hard work!

File: tests/test_acm/test_acm.py
Comment (bblommers) [inline tests/test_acm/test_acm.py:502]: Workaround as pytest doesn't seem to like the custom fixture argument ('frozen_time')

**Review by bblommers — APPROVED** [review]

**Comment by bblommers** [comment]
Would you mind having a look as well, @spulec? (When you have a few hours..)
Looks all good to me.

**Comment by spulec** [comment]
Looks great; thanks everyone for the work!

**Pull request merged**
